onnxruntime/core/providers/cuda/math/cumsum.cc

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "cumsum.h"
+#include "cumsum_impl.h"
+#include "core/providers/cpu/math/cumsum.h"
+#include "core/providers/cpu/tensor/utils.h"
+#include "core/providers/common.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+ONNX_OPERATOR_KERNEL_EX(
+    CumSum,
+    kOnnxDomain,
+    11,
+    kCudaExecutionProvider,
+    KernelDefBuilder()
+        .InputMemoryType<OrtMemTypeCPUInput>(1)  // 'axis' needs to be on CPU
+        .TypeConstraint("T", std::vector<MLDataType>{
+                                 DataTypeImpl::GetTensorType<int32_t>(),
+                                 DataTypeImpl::GetTensorType<int64_t>(),
+                                 DataTypeImpl::GetTensorType<uint32_t>(),
+                                 DataTypeImpl::GetTensorType<uint64_t>(),
+                                 DataTypeImpl::GetTensorType<float>(),
+                                 DataTypeImpl::GetTensorType<double>(),
+                                 DataTypeImpl::GetTensorType<MLFloat16>()})
+        .TypeConstraint("T2", std::vector<MLDataType>{DataTypeImpl::GetTensorType<int32_t>(), DataTypeImpl::GetTensorType<int64_t>()}),
+    CumSum);
+
+Status CumSum::ComputeInternal(OpKernelContext* ctx) const {
+  const Tensor* input = ctx->Input<Tensor>(0);                             // input tensor
+  const auto rank = static_cast<int64_t>(input->Shape().NumDimensions());  // the rank of the input/output
+  const Tensor* axis_tensor = ctx->Input<Tensor>(1);                       // axis input tensor
+
+  int64_t axis;
+  auto status = cumsum_op::GetAxis(axis_tensor, rank, axis);
+  if (!status.IsOK())
+    return status;
```

Comment:
ORT_THROW_IF_ERROR(cusum_op::GetAxis(...))

Reply:
Fixed.

Reply:
After a second thought, maybe ORT_RETURN_IF_ERROR is better.

---
In reply to: [357868531](https://github.com/microsoft/onnxruntime/pull/2647#discussion_r357868531) [](ancestors = 357868531)